ee/backend/app/src/main/kotlin/io/tolgee/ee/component/slackIntegration/notification/SlackAutomationMessageSender.kt

```diff
@@ -0,0 +1,275 @@
+import io.tolgee.ee.component.slackIntegration.data.SlackMessageDto
+import io.tolgee.ee.component.slackIntegration.data.SlackRequest
+import io.tolgee.ee.component.slackIntegration.notification.messageFactory.SlackImportMessageFactory
+import io.tolgee.ee.component.slackIntegration.notification.messageFactory.SlackOnKeyAddedMessageFactory
+import io.tolgee.ee.component.slackIntegration.notification.messageFactory.SlackTooManyTranslationsMessageFactory
+import io.tolgee.ee.component.slackIntegration.notification.messageFactory.SlackTranslationChangeMessageFactory
+import io.tolgee.ee.service.slackIntegration.SavedSlackMessageService
+import io.tolgee.model.slackIntegration.SavedSlackMessage
+import io.tolgee.model.slackIntegration.SlackConfig
+import io.tolgee.service.language.LanguageService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationContext
+import org.springframework.context.annotation.Lazy
+import org.springframework.stereotype.Component
+import kotlin.collections.forEach
+import kotlin.collections.plus
+
+@Lazy
+@Component
+class SlackAutomationMessageSender(
+  private val applicationContext: ApplicationContext,
+  private val savedSlackMessageService: SavedSlackMessageService,
+  private val blocksProvider: SlackNotificationBlocksProvider,
+  private val slackOperations: SlackChannelMessagesOperations,
+  private val languageService: LanguageService,
+  private val slackOnKeyAddedMessageFactory: SlackOnKeyAddedMessageFactory,
+  private val slackTranslationChangeMessageFactory: SlackTranslationChangeMessageFactory,
+  private val slackImportMessageFactory: SlackImportMessageFactory,
+  private val slackTooManyTranslationsMessageFactory: SlackTooManyTranslationsMessageFactory,
+) {
+  val logger: Logger by lazy {
+    LoggerFactory.getLogger(javaClass)
+  }
+
+  companion object {
+    const val MAX_MESSAGES_TO_UPDATE = 5
+    const val MAX_NEW_MESSAGES_TO_SEND = 5
+  }
+
+  fun sendMessageOnTranslationSet(
+    slackConfig: SlackConfig,
+    data: SlackRequest,
+  ) {
+    val context = createContext(slackConfig, data)
+    if (context.isBigOperation) {
+      logger.debug("Too many translations to send message, sending only one message")
+      val messageDto = slackTooManyTranslationsMessageFactory.createMessageIfTooManyTranslations(context)
+      sendRegularMessageWithSaving(messageDto, context)
+      return
+    }
+
+    val messagesDto = slackTranslationChangeMessageFactory.createTranslationChangeMessages(context)
+    val savedMessagesByKey = savedSlackMessageService.findAll(messagesDto, context.slackConfig.id).groupBy { it.keyId }
+
+    val messagesToUpdate = mutableMapOf<Long, MutableList<SlackMessageDto>>()
+    val messagesToCreate = mutableListOf<SlackMessageDto>()
+
+    for (message in messagesDto) {
+      if (!isExplicitlySubscribedToAnyUpdatedLanguage(message, context.slackConfig)) {
```

Comment:
I guess some tests don't pass because of this part.

The logic should be like this:
 - When I am subscribed to some languages explicitely, I always get new message on change
 - When I am subscribed with non-specific language subscription, I get messages updated only when something change
   - That's because specific language translators/reviewers want to see all the changes. Product managers and others just need to check it was changed but they don't need to be always notified
 - When there is more then treshold messages I am notified about count only and I get a button to see it in Tolgee